clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -7706,7 +7706,7 @@ ExprResult Sema::CheckTemplateArgument(NonTypeTemplateParmDecl *Param,
     // FIXME: The language rules don't say what happens in this case.
     // FIXME: We get an opaque dependent type out of decltype(auto) if the
     // expression is merely instantiation-dependent; is this enough?
-    if (CTAK == CTAK_Deduced && Arg->isTypeDependent()) {
+    if (Arg->isTypeDependent()) {
```

Comment:
I think this makes reasonable sense, I note that this covers the  CTAK==CTAK_Specified case, but I would like a test where it is an array-bound so that we can ensure we're getting that right as well.

Reply:
After looking into the code a bit more, I think this won't happen since the type of `Arg` can't be dependent if the condition `CTAK == CTAK_DeducedFromArrayBound` holds. Because it has been deduced as a certain expression at the moment.